crates/rome_js_parser/src/syntax/typescript/types.rs

```diff
@@ -1294,35 +1516,25 @@ pub(crate) fn parse_ts_type_arguments_in_expression(p: &mut JsParser) -> ParsedS
     .unwrap_or(Absent)
 }
 
-#[inline]
-pub fn can_follow_type_arguments_in_expr(cur_kind: JsSyntaxKind) -> bool {
+fn can_follow_type_arguments_in_expr(p: &mut JsParser) -> bool {
+    let cur_kind = p.cur();
+    match cur_kind {
+        // These tokens can follow a type argument list in a call expression.
+        T!['('] | BACKTICK | EOF => true,
+        // A type argument list followed by `<` never makes sense, and a type argument list followed
+        // by `>` is ambiguous with a (re-scanned) `>>` operator, so we disqualify both. Also, in
+        // this context, `+` and `-` are unary operators, not binary operators.
+        T![<] | T![>] | T![+] | T![-] => false,
+        // We favor the type argument list interpretation when it is immediately followed by
+        // a line break, a binary operator, or something that can't start an expression.
+        _ => p.has_preceding_line_break() || is_binary_operator(p) || !is_at_expression(p),
+    }
+}
+
+fn is_binary_operator(p: &mut JsParser) -> bool {
```

Comment:
Nit

```suggestion
fn is_at_binary_operator(p: &JsParser) -> bool {
```

I recommend moving this to `expr.rs` next to the binary expression parsing

Reply:
moved